src/proj.h

```diff
@@ -0,0 +1,202 @@
+union PJ_TRIPLET;
+typedef union PJ_TRIPLET PJ_TRIPLET;
+
+/* Data type for generic geodetic 3D data plus epoch information */
+union PJ_SPATIOTEMPORAL;
+typedef union PJ_SPATIOTEMPORAL PJ_SPATIOTEMPORAL;
+
+#ifndef PROJECTS_H
+/* Data type for projection/transformation information */
+struct PJconsts;
+typedef struct PJconsts PJ;         /* the PJ object herself */
+
+/* Omega, Phi, Kappa: Rotations */
+typedef struct {double o, p, k;}  OPK;
+
+/* Easting, Northing, and some kind of height (orthometric or ellipsoidal) */
+typedef struct {double e, n, h;}  ENH;
+
+/* Geodetic spatiotemporal coordinate types */
+typedef struct { double   x,   y,  z, t; }  XYZT;
+typedef struct { double   e,   n,  h, t; }  ENHT;
+typedef struct { double   u,   v,  w, t; }  UVWT;
+typedef struct { double lam, phi,  z, t; }  LPZT;
+
+/* Classic proj.4 pair/triplet types */
+typedef struct { double   u,   v; }  UV;
+typedef struct { double   x,   y; }  XY;
+typedef struct { double lam, phi; }  LP;
+
+typedef struct { double   x,   y,  z; }  XYZ;
+typedef struct { double   u,   v,  w; }  UVW;
+typedef struct { double lam, phi,  z; }  LPZ;
+
+/* Ancillary pairs and triplets for geodetic computations */
+
+/* Geoid undulation and deflections of the vertical */
+typedef struct { double   N,   z,  e; }  PJ_ANC_GEOIDAL;
+
+/* Ellipsoidal parameters */
+typedef struct { double   a,   f; }  PJ_ANC_ELLIPSOIDAL;
+#endif  /* ndef PROJECTS_H */
+
+union PJ_SPATIOTEMPORAL {
+#ifndef PROJECTS_H
+    XYZT xyzt;
+    UVWT uvwt;
+    ENHT enht;
+    LPZT lpzt;
+    ENH  enh;
+#endif
+    double v[4]; /* Who cares - it's just a vector! */
+    XYZ  xyz;
+    UVW  uvw;
+    LPZ  lpz;
+    XY   xy;
+    UV   uv;
+    LP   lp;
+};
+
+
+
+/* Avoid preprocessor renaming and implicit type-punning: Use a union to make it explicit */
+union PJ_TRIPLET {
+#ifndef PROJECTS_H
+    OPK    opk;
+    ENH    enh;
+    PJ_ANC_GEOIDAL     nze;
+    PJ_ANC_ELLIPSOIDAL ell;
+#endif
+    double v[3]; /* Who cares - it's just a vector! */
+    XYZ    xyz;
+    LPZ    lpz;
+    UVW    uvw;
+    XY     xy;
+    LP     lp;
+    UV     uv;
+};
+
+struct OBSERVATION {
+    PJ_SPATIOTEMPORAL coo;
+    PJ_TRIPLET anc;
+    int id;
+    unsigned int flags;
+};
+
+
+#ifndef PROJECTS_H
+/* Apply transformation to observation - in forward or inverse direction */
+OBSERVATION pj_apply (PJ *P, int direction, OBSERVATION obs);
+
+/* Direction: "+" forward, "-" reverse, 0: roundtrip precision check */
+enum pj_apply_direction {
+    PJ_APPLY_FWD = -1,
+    PJ_APPLY_ROUNDTRIP = 0,
+    PJ_APPLY_INV = 1
+};
+
+int pj_show_triplet (FILE *stream, const char *banner, PJ_TRIPLET point);
+
+/* Constructor for the OBSERVATION object */
+OBSERVATION pj_observation (
+    double x, double y, double z, double t,
+    double o, double p, double k,
+    int id, unsigned int flags
+);
+
+#ifndef TODEG
+#define TODEG(rad)  ((rad)*180/M_PI)
+#endif
+#ifndef TORAD
+#define TORAD(deg)  ((deg)*M_PI/180)
```

Comment:
Would you mind using 180.0 for these two macros?  I seem to remember Visual C++ giving warnings about implicit typecasting.